e2e-tests/src/test/java/gov/nasa/jpl/aerie/e2e/ConstraintsTests.java

```diff
@@ -442,25 +442,26 @@ void compilationFailsOutdatedSimulationSimDatasetId() throws IOException {
       final var message = exception.getMessage().split("\"message\":\"")[1].split("\"}]")[0];
       // Hasura strips the cause message ("Assumption falsified -- mission model for existing plan does not exist")
       // from the error it returns
-      if (!message.equals("input mismatch exception")) {
+      if (!message.equals("not a valid json response from webhook")) {
```

Comment:
Yeah I was confused by that as well, but I double checked recreating the error in the UI and I see the same error as well. I'm not sure if we're getting deeper into the constraint code checking or if something else is going on. I can get the full stack in a bit!

Reply:
I took a quick dive into it and it looks like what happened is that the action will throw an Error on L167/L169 which isn't getting caught and serialized into a proper response by the Merlin Bindings. This causes it to have an invalid format when it returns to Hasura. Prior to this PR, it would be instead thrown as an `InputMismatchException`, which is properly handled. I'm not sure what change caused this, however.

Reply:
Summary for posterity: Prior to this  PR, codegen would include unrelated external dataset resources, causing compilation to pass but evaluation of the constraint to fail (see DiscreteResource L29 for the original error message), but now compilation is failing because codegen (correctly) is not include those excess resources